snuba/query/project_extension.py

```diff
@@ -115,6 +127,15 @@ def do_post_processing(
                     query.add_conditions(
                         [(["assumeNotNull", ["group_id"]], "NOT IN", exclude_group_ids)]
                     )
+                    query.add_condition_to_ast(
+                        not_in_condition(
+                            None,
+                            FunctionCall(
+                                None, "assumeNotNull", (Column(None, "group_id", None),)
+                            ),
+                            [Literal(None, p) for p in exclude_group_ids],
+                        )
+                    )
```

Comment:
I wanted to avoid these kinds of duplication but none of the alternative seemed appealing
- pass the old syntax to the query and let it figure it out to generate the new one in parallel. This has two pretty nasty downsides: the query needs to parse the old syntax into the new one, which is impossible because the parser depends on the query, not vice versa. Also it does not set the expectations that the users will have to write the new syntax.
- pass the new syntax to query and let it figure out. It means writing a new formatter that formats the new conditions into conditions for the old query syntax.

So, in a separate PR I will experimenting in adding a wrapper to the Query that takes the new syntax and formats it into old and applies both to the existing query object so we would be passing only the new syntax.